apps/src/music/blockly/blockUtils.js

```diff
@@ -134,9 +135,9 @@ function restoreBlockDefinitions() {
 }
 
 // A helper function to generate the code for a function call to play sounds sequentially.
-function simple2FunctionCallGenerator(functionName) {
+function simple2FunctionCallGenerator(functionName, functionCallBllockId) {
   return `
-    Sequencer.startFunctionContext('${functionName}');
+    Sequencer.startFunctionContext('${functionName}', '${functionCallBllockId}');
```

Comment:
Do we need to add this to `AdvancedSequencer` too?

Reply:
I think this implementation of function context only exists in the `Simple2` model, though @mikeharv might know more.

Reply:
Brendan is correct. This particular condition only works in Simple2 because only Advanced does have a concept of function contexts. In Advanced mode, we can check that a sound was played by a block contained in a function definition, but we don't currently support checking the specific function name. I've been told this isn't a blocker for CSD.